js/src/views/Dapps/dappsFetcher.js

```diff
@@ -0,0 +1,271 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import BigNumber from 'bignumber.js';
+import { pick, range, uniq } from 'lodash';
+import EventEmitter from 'eventemitter3';
+
+import Contracts from '~/contracts';
+import { hashToImageUrl } from '~/redux/util';
+import { bytesToHex } from '~/api/util/format';
+
+import builtinApps from './builtin.json';
+
+const BUILTIN_APPS_KEY = 'BUILTIN_APPS_KEY';
+let dappsFetcherInstance = null;
+
+export default class DappsFetcher extends EventEmitter {
```

Comment:
The idea was more to have a store for data (that gets persisted throughout the session) and one for the UI, which is a MobX store with observers & all. The idea is that we might not want to preserve MobX stores throughout the app since they don't old any valuable data if the component is not mounted.